tools/clang/test/HLSLFileCheck/hlsl/workgraph/test_EmitDxilFunctionProps_MDVals.hlsl

```diff
@@ -0,0 +1,28 @@
+// RUN: %dxc -fcgl -T lib_6_8 %s | FileCheck %s
+//
+// Verify that the MDVals data structure inside EmitDxilFunctionProps in 
+// DxilMetadataHelper.cpp doesn't get any out-of-bounds assignments.
+
+// CHECK: !36
```

Comment:
What is this verifying?  I don't think this is sufficient, and numeric metadata nodes are just unnamed nodes with automatically assigned indices.  It could be anything.  In the future, metadata could be written and this would probably not be checking what you think it is.

Reply:
Originally, when the MDVals data structure was capped at 35 entries, and entries were entered via index assignment, an out-of-bounds assignment would occur because this specific path would require more than 35 values to be entered into MDVals. When the data structure was changed (as well as the insertion method to push_back), there was no accompanying test. Ideally, we would want a regression test that passes when push_back is used, and fails in the old code where the size was capped and index assignment was used. 
I thought this test would be sufficient because it verifies that the data structure has at least 36 entries, which was impossible with the old code, since an out of bounds assignment would have prevented successful compilation.

Reply:
The indices of metadata nodes aren't 1:1 with the indices of that particular array.  These indices are automatically assigned for each unnamed metadata node, of which there are others besides ones used in the array, and not every array element refers to another unique metadata node either.

Reply:
Given that the old test, "D:\DXC\tools\clang\test\HLSLFileCheck\hlsl\workgraph\test_EmitDxilFunctionProps_MDVals.hlsl", already serves as a regression test for the change of the insertion method, I've removed the redundant test. There is no way to trigger the out-of-bounds behavior without the definition of a node shader, and so an explanation has been added to the old test on how it serves its extra purpose.